src/scheduler/job_info.c

```diff
@@ -3426,11 +3429,42 @@ find_jobs_to_preempt(status *policy, resource_resv *hjob, server_info *sinfo, in
 
 		update_universe_on_end(npolicy, pjob,  "S", NO_ALLPART);
 		rjobs_count--;
+		/* Check if any of the previously preempted job increased its preemption priority */
+		for (ind = 0; pjobs[ind] != NULL; ind++) {
```

Comment:
When we find a preemption candidate inside select_index_to_preempt(), is there a way to check how much its preempt priority will increase if it got preempted? If we could determine that, then we can just not select such a job inside select_index_to_preempt() and we wouldn't need to check stuff here.

Reply:
We can add that check but that check will be kind of similar to what we have right now. We will have to end the job in a simulated universe and then check its preemption priority. 
Even if we have this check, We can not predict if its preemption priority will move if any of the other running jobs get preempted.